### DISCUSSION TITLE
Can I use ClassValidator decorators in the entities?

### DISCUSSION THREAD

**Discussion by bill-o-more** [mikro-orm/mikro-orm#4522] [Q&A] [answered]
Hi,
Currently migrating a project from TypeOrm, and have a question (sorry if this sort of issues don't belong on issue tracker).

I read #720, but couldn't determine if I can use ClassValidator inside the entities themselves, not DTOs? I'll use [BaseEntity](https://mikro-orm.io/docs/defining-entities#using-custom-base-entity) and want to put some constraints on the values I add (e.g. make the phone numbers match certain pattern). I don't need it to handle collections, only plain values

**Comment by B4nan** [comment, marked as answer]
class-validator is fine, the problem is class-transformer, as that package has no idea of how to deal with the ORM-specific mapping and identity map in general

**Reply by bill-o-more** [reply]
That's great news! So if I don't need transformer, I can just use something like 
```typescript
@Property()
@Length(13, 13)
@Matches(MOBILE_PHONE_REGEX)
mobilePhone: string;
```
and be sure that no string that doesn't match the regex won't get into the DB?

**Reply by B4nan** [reply]
You will need to trigger the validation somewhere, eg in some hook like `beforeCreate`
